app/services/degrees/create_from_apply.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Degrees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(attributes:)
+      @attributes = attributes
+    end
+
+    def call
+      degree
+    end
+
+  private
+
+    attr_reader :attributes
+
+    def degree
+      Degree.new(params)
+    end
+
+    def params
+      @params ||= common_params.merge(degree_params)
+    end
+
+    def degree_params
+      uk_degree? ? uk_degree_params : non_uk_degree_params
+    end
+
+    # TODO: Check that each of degree and subject is in our
+    # Dttp codeset mappings before saving.
+    def common_params
+      {
+        subject: attributes["subject"],
+        graduation_year: attributes["award_year"],
+      }
+    end
+
+    # TODO: Check that institution is in our
+    # Dttp codeset mappings before saving.
+    def uk_degree_params
+      {
+        locale_code: Trainee.locale_codes[:uk],
+        uk_degree: attributes["qualification_type"],
+        institution: institution,
+        grade: attributes["grade"],
+      }
+    end
+
+    def non_uk_degree_params
+      {
+        locale_code: Trainee.locale_codes[:non_uk],
+        # TODO: This should be the comparable_uk_degree` from Apply
+        non_uk_degree: attributes["non_uk_qualification_type"],
+        country: country,
```

Comment:
Worth noting that this information could be unavailable for non-degree qualifications 
https://github.com/DFE-Digital/apply-for-teacher-training/blob/ea87f086b9e55a21248fb53143c1f36689b64735/spec/presenters/hesa_qualification_fields_presenter_spec.rb#L33-L46

However, would be worth double checking with apply as the API docs have no mention of any `hesa_*` fields against qualifications https://sandbox.apply-for-teacher-training.service.gov.uk/register-api#get-applications